lms/djangoapps/ccx/views.py

```diff
@@ -422,20 +446,17 @@ def ccx_invite(request, course, ccx=None):
     identifiers = _split_input_list(identifiers_raw)
     auto_enroll = True
     email_students = True if 'email-students' in request.POST else False
-    for identifier in identifiers:
-        user = None
-        email = None
-        try:
-            user = get_student_from_identifier(identifier)
-        except User.DoesNotExist:
-            email = identifier
-        else:
-            email = user.email
-        try:
-            validate_email(email)
-            course_key = CCXLocator.from_course_locator(course.id, ccx.id)
-            email_params = get_email_params(course, auto_enroll, course_key=course_key, display_name=ccx.display_name)
-            if action == 'Enroll':
+    course_key = CCXLocator.from_course_locator(course.id, ccx.id)
+    ccx_course_overview = CourseOverview.get_from_id(course_key)
+    email_params = get_email_params(course, auto_enroll, course_key=course_key, display_name=ccx.display_name)
+
+    if action == 'Enroll':
+        for identifier in identifiers:
```

Comment:
>@giocalitri why not do this to save code duplication, i mean using for loop for both enroll and unenroll?
```
for identifier in identifiers:
  if action == 'Enroll':
     if ccx has not filled and user is valid
        add user
  else if action == "Unenroll":
     unenroll user
```

I think this approach is wrong: there can be only one `action` per submission and it is useless to check it at each loop.